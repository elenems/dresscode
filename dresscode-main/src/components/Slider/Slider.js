import React, { PureComponent } from "react";
import AliceCarousel from "react-alice-carousel";
import "react-alice-carousel/lib/alice-carousel.css";
import Icon from "../../utils/Icon";
import PropTypes from "prop-types";

const res = {
  0: {
    items: 1
  },
  500: {
    items: 2
  },
  768:{
    items: 3
  },
  1024: {
    items: 4
  },
  1920: {
    items: 5
  }
};

class Slider extends PureComponent {
  state = {
    currentIndex: 0
  };

  onSlideChanged = e => this.setState({ currentIndex: e.item });

  slideNext = () =>
    this.setState({ currentIndex: this.state.currentIndex + 1 });

  slidePrev = () =>
    this.setState({ currentIndex: this.state.currentIndex - 1 });

  render() {
    return this.props.items.length ? (
      <div style={{ position: "relative" }}>
        <AliceCarousel
          items={this.props.items}
          buttonsDisabled={true}
          dotsDisabled={true}
          onSlideChanged={this.onSlideChanged}
          slideToIndex={this.state.currentIndex}
          responsive={res}
          mouseDragEnabled
        />
        <div className="alice-carousel__prev-btn">
          <button
            className="alice-carousel__prev-btn-wrapper"
            onClick={() => this.slidePrev()}
          >
            <Icon color="white" name="left-arrow" />
          </button>
        </div>

        <div className="alice-carousel__next-btn">
          <button
            className="alice-carousel__next-btn-wrapper"
            onClick={() => this.slideNext()}
          >
            <Icon color="white" name="right-arrow" />
          </button>
        </div>
      </div>
    ) : null;
  }
}

Slider.propTypes = {
  items: PropTypes.array
};

export default Slider;
