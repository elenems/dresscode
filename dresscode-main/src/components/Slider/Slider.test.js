import React from "react";
import { shallow } from "enzyme";
import Slider from "./Slider";

describe("Slider", () => {
  let props;
  beforeEach(() => {
    props = {
      items: [{}]
    };
  });
  it("Mathces snapshot", () => {
    const wrapper = shallow(<Slider {...props} />).debug();
    expect(wrapper).toMatchSnapshot();
  });

  it("Changes index on slide change", () => {
    const e = {
      item: 1
    };
    const wrapper = shallow(<Slider {...props} />);
    wrapper.instance().onSlideChanged(e);
    expect(wrapper.state().currentIndex).toEqual(1);
  });

  it("Increments index on next slide", () => {
    const wrapper = shallow(<Slider {...props} />);
    wrapper.instance().slideNext();
    expect(wrapper.state().currentIndex).toBe(1);
  });

  it("Decrements index on prev slide", () => {
    const wrapper = shallow(<Slider {...props} />);
    wrapper.instance().slidePrev();
    expect(wrapper.state().currentIndex).toBe(-1);
  });

  it("Renders nothing if items missing", () => {
    props.items = [];
    const wrapper = shallow(<Slider {...props} />);
    expect(wrapper.get(0)).toBe(null);
  });

  it("Can clicks slidePrev", () => {
    const wrapper = shallow(<Slider {...props} />);
    const index = wrapper.state("currentIndex");
    wrapper
      .find(".alice-carousel__prev-btn")
      .children("button")
      .simulate("click");
    expect(wrapper.state("currentIndex")).toBe(index - 1);
  });

  it("Can clicks slideNext", () => {
    const wrapper = shallow(<Slider {...props} />);
    const index = wrapper.state("currentIndex");
    wrapper
      .find(".alice-carousel__next-btn")
      .children("button")
      .simulate("click");
    expect(wrapper.state("currentIndex")).toBe(index + 1);
  });
});
