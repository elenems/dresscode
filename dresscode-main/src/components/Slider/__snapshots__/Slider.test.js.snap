// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Mathces snapshot 1`] = `
"<div style={{...}}>
  <AliceCarousel items={{...}} buttonsDisabled={true} dotsDisabled={true} onSlideChanged={[Function]} slideToIndex={0} responsive={{...}} mouseDragEnabled={true} stagePadding={{...}} duration={250} startIndex={0} autoPlay={false} infinite={true} showSlideInfo={false} swipeDisabled={false} autoPlayInterval={250} fadeOutAnimation={false} playButtonEnabled={false} autoPlayDirection=\\"ltr\\" keysControlDisabled={false} disableAutoPlayOnAction={false} stopAutoPlayOnHover={true} preventEventOnTouchMove={false} autoHeight={false} />
  <div className=\\"alice-carousel__prev-btn\\">
    <button className=\\"alice-carousel__prev-btn-wrapper\\" onClick={[Function: onClick]}>
      <Memo(Icon) color=\\"white\\" name=\\"left-arrow\\" />
    </button>
  </div>
  <div className=\\"alice-carousel__next-btn\\">
    <button className=\\"alice-carousel__next-btn-wrapper\\" onClick={[Function: onClick]}>
      <Memo(Icon) color=\\"white\\" name=\\"right-arrow\\" />
    </button>
  </div>
</div>"
`;
